Cargo.toml

```diff
@@ -39,7 +39,8 @@ ansi_term = { version = "0.11",  optional = true }
 
 [dev-dependencies]
 regex = "1"
-lazy_static = "1"
+# lazy_static 1.2 requires Rust 1.24.1+
+lazy_static = "~1.1"
```

Comment:
An alternative approach here is to pin `lazy_static` in your lock file in CI. For example: https://github.com/rust-lang/regex/blob/770edd59b2c4d84dc40231e8648eeb65b2f0d94a/ci/script.sh#L8-L17

The benefit of pinning in the lock file is that consumers of clap on newer versions of Rust can just use `lazy_static 1.2`, but I think with the constraint in `Cargo.toml` like you have here, you'll either wind up with two versions of `lazy_static` in the build or the _just_ the older version of `lazy_static`. (I can't remember how Cargo behaves here, but I think it's the latter behavior since they are semver compatible releases.)

Reply:
How does that work with non-CI tests (e.g. if I run `cargo +1.21.0 test` locally)?

Also, I may be wrong on this, but my understanding was that `dev-dependencies` wouldn't get propagated to consumers of clap and were only used for examples/tests/etc.; is that incorrect?

Reply:
`cargo +1.21.0 test` would fail. Ultimately, this is a judgment call, and it's possible that by virtue of being a `dev-dependency` this won't have the impact I claimed above. I'm not sure.

Reply:
I just created a fresh project with this in `Cargo.toml`:

```
[dependencies]
lazy_static = "1"

[dependencies.clap]
git = "https://github.com/Elarnon/clap.git"
branch = "lazy_static_1.1"
```

and ended up with only `lazy_static` 1.2 in `Cargo.lock`, so it seems this should be fine for `dev-dependencies`.